components/script/dom/websocket.rs

```diff
@@ -608,13 +608,17 @@ impl Runnable for MessageReceivedTask {
                             blob.to_jsval(cx, message.handle_mut());
                         }
                         BinaryType::Arraybuffer => {
-                            let len = data.len() as uint32_t;
-                            let buf = JS_NewArrayBuffer(cx, len);
+                            let len = data.len() as u32;
+                            rooted!(in(cx) let mut array_buffer = ptr::null_mut());
+                            assert!(Uint8Array::create(cx, len, Some(data.as_slice()),
+                                    array_buffer.handle_mut()).is_ok());
+
                             let mut is_shared = false;
-                            let buf_data: *mut uint8_t = JS_GetArrayBufferData(buf, &mut is_shared, ptr::null());
+                            let buf_data: *mut uint8_t =
+                                JS_GetUint8ArrayData(*array_buffer, &mut is_shared, ptr::null());
                             assert!(!is_shared);
                             ptr::copy_nonoverlapping(data.as_ptr(), buf_data, len as usize);
```

Comment:
All of the code to fill the buffer here is now unnecessary, since that's handled in by the `create` method.